File: Spring-backend/Springbackend/src/main/java/com/dravicenne/backend/users/medecin/Medecin.java
package com.dravicenne.backend.users.medecin;

import com.dravicenne.backend.users.User;

import javax.persistence.Entity;


@Entity(name = "medecin")
public class Medecin extends User {

    private String specialite;
    private String autreSpecialites;

    public Medecin() {
    }

    public Medecin(String nom, String prenom, String email, String password, String confPassword, String status, String specialite, String autreSpecialites) {
        super(nom, prenom, email, password, confPassword, status);
        this.specialite = specialite;
        this.autreSpecialites = autreSpecialites;
    }

    public String getSpecialite() {
        return specialite;
    }

    public void setSpecialite(String specialite) {
        this.specialite = specialite;
    }

    public String getAutreSpecialites() {
        return autreSpecialites;
    }

    public void setAutreSpecialites(String autreSpecialites) {
        this.autreSpecialites = autreSpecialites;
    }
}

File: Angular-frontend/src/app/Models/Medecin/medecin.ts
import { User } from '../User/user';


export class Medecin extends User{
  public cin: string;
  public specialite: string;

  // tslint:disable-next-line: max-line-length
  constructor(nom: string, prenom: string, ville: string, phone: string, email: string, password: string, Cpassword: string, cin: string, specialite: string)
  {
    super(nom, prenom, ville, phone, email, password, Cpassword);
    this.cin = cin;
    this.specialite = specialite;
  }

  public getCin(): string
  {
    return this.cin;
  }

  public getSpecialite(): string
  {
    return this.specialite;
  }

}


